Design/Workbench/design-layout.md

```diff
@@ -0,0 +1,149 @@
+Summary
+=======
+
+This document describes the layout and high-level details of the new matplotlib-based workbench that will replace MantidPlot.
+
+Workflow
+========
+
+The new workbench will be developed on a long-running feature branch within the main [`mantid`][mantidrepo] repository. The exception to this is the new directory
+structure described below for the graphical components. These changes will be made on the `master` branch prior to any work starting on the workbench. This
+will minimize the effect of merge conflicts on existing components that are going to be reused.
+
+The advantage here is that we can remove the existing MantidPlot on this branch and reuse the name for the new workbench.
+
+
+Directory Structure
+===================
+
+The following diagram indicates the proposed directory structure for the graphical components.
+
+```
+mantid.git
+   |-- Framework
+   |-- MantidPlot
+   |-- qt
+       |-- applications
+       |   |-- mantidworkbench
+       |-- paraview_ext
+       |-- resources
+       |-- scientific_interfaces
+       |   |-- CMakeLists.txt
+       |   |-- Diffraction
+       |   |   |-- PowderDiffractionReduction
+       |   |       |-- Powder_Diffraction_Reduction.py
+       |   |       |-- Powder_Diffraction_Reduction
+       |   |       |   |-- ...
+       |   |-- Indirect
+       |   |   |-- Common
+       |   |   |-- Corrections
+       |   |   |-- DataAnalysis
+       |   |-- Muon
+       |   |   |-- ALC
+       |   |   |-- DataAnalysis
+       |   |-- Reflectometry
+       |   |   |-- ISISReflectometry
+       |   |       |-- ISIS_Reflectometry.py
+       |   |       |-- ISIS_Reflectometry
+       |   |       |   |-- ...
+       |-- python
+       |   |-- mantidqt # python mantidqt module
+       |       |-- reduction_gui # Python reduction gui framework
+       |       |-- widgets # would contain combine qtwidgetscommon.dll with sip exports to give mantidqt.widgets python library
+       |       |-- __init__.py
+       |-- widgets
```

Comment:
Good point.

I think a worse case scenario would be `master_incremental`. Its code gets clone to `C:\Jenkins\workspace\master_incremental\label\win7\mantid` so that gives us a ~50 character minimum stem. It looks like the `qt\python\mantidqt\reduction_gui` sub directory has the most nesting currently and taking one the files in the leaf directory would take us to ~125 characters at present. This would leave us with a headroom of ~135 characters. I *think* this should be okay even with the deepish directory structure that CMake creates in the build directory.